ui/src/hooks/useFileStream.ts

```diff
@@ -0,0 +1,80 @@
+'use client';
+
+import { QueryKeys } from "@/data-provider/query-service";
+import { TFileIngest, TStreamProgressState } from "@/data-provider/types";
+import { fetchEventSource } from "@microsoft/fetch-event-source";
+import { useQueryClient } from "@tanstack/react-query";
+import { useCallback, useEffect, useState } from "react";
+
+export const useFileStream = () => {
+  const [currentState, setCurrentState] = useState<TStreamProgressState | null>(
+    null,
+  );
+  const [controller, setController] = useState<AbortController | null>(null);
+  const [isStreaming, setIsStreaming] = useState(false);
```

Comment:
So as in just remove the state - because `currentState` is already a state so `isStreaming` will automatically be reassigned on `currentState` change! So right. 

For the second comment - reference the TFileStreamStatus directly right?

Reply:
I updated `useFileStream` to reflect these changes too.